cmake/TorchConfig.cmake.in

```diff
@@ -13,6 +13,27 @@
 # and the following imported targets:
 #
 #   torch
+macro(append_torchlib_if_found)
+  foreach (_arg ${ARGN})
+    find_library(${_arg}_LIBRARY ${_arg} PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    if(${_arg}_LIBRARY)
+      list(APPEND TORCH_LIBRARIES ${${_arg}_LIBRARY})
+    else()
+      message(WARNING "static library ${${_arg}_LIBRARY} not found.")
+    endif()
+  endforeach()
+endmacro()
+
+function(add_whole_archive_flag lib output_var)
```

Comment:
Can you please explain why `add_whole_archive_flag` is a function, but `append_torchlib_if_found` is macro?

Reply:
list append seems not support parent scope, while "set list" seems not intuitive, so use macro instead.